packages/webclient/src/AuthHandler/AzCliAuthHandler.ts

```diff
@@ -1,77 +1,100 @@
+        let azCloudDetails = JSON.parse(await this.executeAzCliCommand('cloud show'));
+
+        this._subscriptionId = azAccountDetails['id'];
+        this._cloudSuffixes = azCloudDetails['suffixes'];
+        this._cloudEndpoints = azCloudDetails['endpoints'];
     }
 
     public async getToken(force?: boolean, args?: string[]): Promise<string> {
         if(!this._token || force) {  
             try {
-                let azAccessToken = JSON.parse(await executeAzCliCommand('account get-access-token', !!args ? args : []));
+                let azAccessToken = JSON.parse(await this.executeAzCliCommand('account get-access-token', !!args ? args : []));
+                console.log(`::add-mask::${azAccessToken}`);
                 this._token = azAccessToken['accessToken'];
             }
             catch(error) {
                 console.log('Failed to fetch Azure access token');
                 throw error;
             }          
         }
+
         return this._token;
     }
-}
+    
+    public static async getAzCliPath(): Promise<string> {
+        if (!this._azCliPath) {
+            this._azCliPath = await io.which('az', true);
+        }
+
+        return this._azCliPath;
+    }
 
-export async function initialize() {  
-    azPath = await io.which("az", true);
-    azCloudDetails = JSON.parse(await executeAzCliCommand('cloud show'));
-}
+    public async executeAzCliCommand(command: string, args?: string[]): Promise<string> {
+        let azCliPath = await AzCliAuthHandler.getAzCliPath();
+        let stdout = '';
+        let stderr = '';
 
-async function executeAzCliCommand(command: string, args?: string[]): Promise<string> {
-    let stdout = '';
-    let stderr = '';
-   
-    let code = await exec.exec(`"${azPath}" ${command}`, args, {
-        silent: true,
-        listeners: {
-            stdout: (data: Buffer) => {
-                stdout += data.toString();
-            },
-            stderr: (data: Buffer) => {
-              stderr += data.toString();
-            }
+        try {
+            core.debug(`"${azCliPath}" ${command}`);
+            await exec.exec(`"${azCliPath}" ${command}`, args, {
+                silent: true, // this will prevent priniting access token to console output
+                listeners: {
+                    stdout: (data: Buffer) => {
+                        stdout += data.toString();
+                    },
+                    stderr: (data: Buffer) => {
+                      stderr += data.toString();
+                    }
+                }
+            });
         }
-    }); 
-
-    if(code != 0) {
-        core.debug('Failed to fetch Azure access token');
-        throw new Error(stderr);
+        catch(error) {
+            console.log('Failed to fetch Azure access token');
+            throw new Error(stderr);
+        }
+        
+        return stdout;
     }
 
-    return stdout;
+    private _token: string = '';

```

Comment:
suggestion: move this to top of class declaration

Reply:
We have been following this pattern in all our ts and tsx files in AzureDevOps repo.